wicker/core/datasets.py

```diff
@@ -151,55 +249,55 @@ def get_folder_size(bucket, prefix):
 
         # need to know which columns are heavy pntr columns we go to for
-        # byte adding
+        # byte adding, parse which are heavy pointers based off metadata
+        # baked into schema
         schema = self.schema()
         heavy_pointer_cols = []
         for col_name in schema.get_all_column_names():
-            if schema.get_column(col_name).is_heavy_pointer:
+            column = schema.get_column(col_name)
+            if column is not None and column.is_heavy_pointer:
                 heavy_pointer_cols.append(col_name)
 
-        # Each individual row only knows which column file it goes to, so we have to
-        # neccesarily parse all rows :( to get the column files. This should be cached
-        # as metadata but that would require re-curating the datasets.
+        # create arrow table for parsing
         print("Creating arrow table")
         arrow_table = self.arrow_table()
 
         buckets_keys = set()
-        worker = ShuffleWorker(storage=self._storage)
+
         print("Processing through heavy pointers")
         for heavy_pntr_col in heavy_pointer_cols:
             print(f"Evaulating {heavy_pntr_col} for column file locations")
+            # Each individual row only knows which column file it goes to, so we have to
+            # neccesarily parse all rows :( to get the column files. This should be cached
+            # as metadata but that would require re-curating the datasets.
             for location_bytes in tqdm.tqdm(arrow_table[heavy_pntr_col].to_pylist()):
                 location = ColumnBytesFileLocationV1.from_bytes(location_bytes)
-                path = worker.s3_path_factory.get_column_concatenated_bytes_s3path_from_uuid(
+                path = self._s3_path_factory.get_column_concatenated_bytes_s3path_from_uuid(
                     location.file_id.bytes, dataset_name=self._dataset_id.name
                 )
                 bucket, key = path.replace("s3://", "").split("/", 1)
                 buckets_keys.add((bucket, key))
 
-        # we use existing shuffle worker here to grab the column file from
-        # know location.
-        def get_file_size_s3(bucket_key: Tuple[str, str]) -> int:
-            s3 = boto3.resource("s3")
-            bucket, key = bucket_key
-            bucket, key = path.replace("s3://", "").split("/", 1)
-            byte_length = s3.Object(bucket, key).content_length
-            return byte_length
-
-        print("Grabbing file information from s3 heads")
-        with ThreadPool() as pool:
-            results = list(tqdm.tqdm(pool.imap(get_file_size_s3, buckets_keys), total=len(buckets_keys)))
-        return sum(results) + par_dir_bytes
+        # pass the data to the multi proc management function
+        buckets_keys_list = list(buckets_keys)
+        column_files_byte_size = get_file_size_s3_multiproc(buckets_keys_list)
+        return column_files_byte_size + par_dir_bytes
```

Comment:
Not sure if the addition here returns the right result if `store_concatenated_bytes_files_in_dataset` and `entire_dataset_parquet_dir_size` are set to `True`? https://github.com/woven-planet/wicker/blob/612f99154454243580f12a189588b3d5dd05bafe/wicker/core/storage.py#L258-L278

Reply:
Yep, what you're suggesting would sum up the parquet files and the COLUMN files for only that partition. So it would be returning essentially the entire parquet but not the entire column files subset. It's because Wicker doesn't keep the meta information for different partitions.

IMO the problem is is that even if you did the n different partitions with no key cutting you would still miss the avro schema because it is not held under any partition in particular. Ideally Wicker is reworked and the s3 object gets ALL partitions but lazily loads one. That is the ideal fix for this but this is a somewhere in between fix that is relevant for copying data to gcloud because we have to have that avro schema on the file path unless we want to do it manually.

Reply:
@pickles-bread-and-butter can you clean up the level of nesting in the storage.py file comment? It looks like things under `/0.0.2` (in both places) should be indented to the right a little bit more, is that right?

> what you're suggesting would sum up the parquet files and the COLUMN files for only that partition. So it would be returning essentially the entire parquet but not the entire column files subset. It's because Wicker doesn't keep the meta information for different partitions.

If `store_concatenated_bytes_files_in_dataset=True` and `entire_dataset_parquet_dir_size=True` I'm not sure that your response is correct. Let's take a look in our meeting.

Reply:
@aalavian Isaak and I discussed this and what we can compute correctly is the size of the current dataset partition. An `S3Dataset` takes a partition name, so its `dataset_size` property should indeed be the size of the partition. We are going to rework this routine a bit so that it reflects this and tag you when this PR is ready for another review.